cmd/influxd/inspect/tsm_verify_test.go

```diff
@@ -0,0 +1,197 @@
+package inspect
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/influxdata/influxdb/v2/tsdb/engine/tsm1"
+)
+
+func TestInvalidChecksum(t *testing.T) {
+	test := newChecksumTest(t, true)
+	defer test.close()
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", test.Path})
+	err := verify.Execute()
+	test.assertError(err)
+}
+
+func TestValidChecksum(t *testing.T) {
+	test := newChecksumTest(t, false)
+	defer test.close()
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", test.Path})
+	err := verify.Execute()
+	test.assertNoError(err)
+
+	out, err := ioutil.ReadAll(b)
+	test.assertNoError(err)
+	test.assert(strings.Contains(string(out), "Broken Blocks: 0 / 1"))
+}
+
+func TestInvalidUTF8(t *testing.T) {
+	test := newUTFTest(t, true)
+	defer test.close()
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", test.Path, "--check-utf8"})
+	err := verify.Execute()
+	test.assertError(err)
+}
+
+func TestValidUTF8(t *testing.T) {
+	test := newUTFTest(t, false)
+	defer test.close()
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", test.Path, "--check-utf8"})
+	err := verify.Execute()
+	test.assertNoError(err)
+
+	out, err := ioutil.ReadAll(b)
+	test.assertNoError(err)
+	test.assert(strings.Contains(string(out), "Invalid Keys: 0 / 1"))
+}
+
+func newUTFTest(t *testing.T, withError bool) *Test {
+	t.Helper()
+
+	dir, err := ioutil.TempDir("", "verify-tsm")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f, err := ioutil.TempFile(dir, "verifytsmtest*"+"."+tsm1.TSMFileExtension)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	w, err := tsm1.NewTSMWriter(f)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer w.Close()
+
+	values := []tsm1.Value{tsm1.NewValue(0, 1.0)}
+	if withError {
+		if err := w.Write([]byte("cpu"), values); err != nil {
```

Comment:
For things that return a single `error` value you can also inline the `NoError` assertion:
```
require.NoError(t, w.Write([]byte("cpu"), values))
```